push-docker-images/action.yml

```diff
@@ -0,0 +1,44 @@
+name: Push Docker Images
+description: Pushes multiple Docker images in parallel.
+
+inputs:
+  images:
+    description: >
+      Docker images to push (local tags or IDs) as a space-separated list.
+      The number of images must match the number of repositories.
+    required: true
+  repos:
+    description: >
+      Docker image repositories (to push the images to) as a space-separated list.
+      The number of repositories must match the number of images.
+    required: true
+  tags:
+    description: Tags to apply to the pushed Docker images as a comma-separated list.
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - shell: bash
+      run: |
+        read -ra images <<< "${{ inputs.images }}"
+        read -ra repos <<< "${{ inputs.repos }}"
+        IFS="," read -ra tags <<< "${{ inputs.tags }}"
+
+        if [ "${#images[@]}" -ne "${#repos[@]}" ]; then
+          echo "Number of images must match the number of repositories."
+          exit 1
+        fi
+
+        for i in "${!images[@]}"; do
+          for tag in "${tags[@]}"; do
+            image="${images[$i]}"
+            repo="${repos[$i]}"
+            target="${repo}:${tag}"
+
+            docker tag $image $target
+            docker push $target &
```

Comment:
If there eare e.g. 3 tags and 3 images, does this `docker push` execute 9 times? I might be wrong, but wouldn't it suffice to `tag` the image `T` times, then `push` once?

Or is it required to `docker push` every `repo:tag` pair?